app/celery/process_ses_receipts_tasks.py

```diff
@@ -36,31 +35,52 @@
 
 @notify_celery.task(bind=True, name='process-ses-result', max_retries=5, default_retry_delay=300)
 @statsd(namespace='tasks')
-def process_ses_results(  # noqa: C901 (too complex 14 > 10)
+def process_ses_results(  # noqa: C901 (too complex 19 > 10)
     self,
     response,
 ):
     current_app.logger.debug('Full SES result response: %s', response)
     try:
         ses_message = json.loads(response['Message'])
+
+        reference = ses_message['mail']['messageId']
+        if not reference:
+            current_app.logger.warning(
+                'SES complaint: unable to lookup notification, messageId (reference) was None | ses_message: %s',
+                ses_message,
+            )
+            return
+
         notification_type = ses_message.get('eventType')
```

Comment:
None of this needs to be in the "try" block.
```suggestion
    ses_message = json.loads(response.get('Message', {}))

    reference = ses_message.get('mail', {}).get('messageId')
    if not reference:
        current_app.logger.warning(
            'SES complaint: unable to lookup notification, messageId (reference) was None | ses_message: %s',
            ses_message,
        )
        return

    notification_type = ses_message.get('eventType')

    try:
```